### PR TITLE
Update examples with New-AzureBatchPool arguments in New-AzureBatchPool

### PR DESCRIPTION
**Pull request by bgklein**
## Description

Updated examples in New-AzureBatchPool.md replacing the deprecated argument OSFamily and version with CloudServicesConfiguration and VirtualMachineConfiguration arguments instead. Created a second basic example  to demonstrate how both configurations worked and then used VirtualMachineConfiguration as a default for the other examples.

## Checklist

- [X] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [X] The title of the PR is clear and informative
- [] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [X] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [NA] If applicable, the changes made in the PR have proper test coverage
- [NA] For public API changes to cmdlets:
    - [NA ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [NA ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/ResourceManager/AzureBatch/Commands.Batch/help/New-AzureBatchPool.md [hunk 1]
```diff
@@ -76,34 +76,48 @@ The **New-AzureBatchPool** cmdlet creates a pool in the Azure Batch service unde
...
 
-### Example 1: Create a new pool using the TargetDedicated parameter set
+### Example 1: Create a new pool using the TargetDedicated parameter set using CloudServiceConfiguration
 ```
-PS C:\>New-AzureBatchPool -Id "MyPool" -VirtualMachineSize "Small" -OSFamily "4" -TargetOSVersion "*" -TargetDedicatedComputeNodes 3 -BatchContext $Context
+PS C:\>$configuration = New-Object -TypeName "Microsoft.Azure.Commands.Batch.Models.PSCloudServiceConfiguration" -ArgumentList @(4,"*")
+PS C:\>New-AzureBatchPool -Id "MyPool" -VirtualMachineSize "Small" -CloudServiceConfiguration $configuration  -TargetDedicatedComputeNodes 3 -BatchContext $Context
+```
+
+### Example 2: Create a new pool using the TargetDedicated parameter set with using VirtualMachineConfiguration
```
Comment (matthchr) [inline src/ResourceManager/AzureBatch/Commands.Batch/help/New-AzureBatchPool.md]: minor remove the "with" in "with using VirutalMachineConfiguration"

File: src/ResourceManager/AzureBatch/Commands.Batch/help/New-AzureBatchPool.md [hunk 2]
```diff
@@ -76,34 +76,48 @@ The **New-AzureBatchPool** cmdlet creates a pool in the Azure Batch service unde
...
+```
+PS C:\$imageReference = New-Object -TypeName "Microsoft.Azure.Commands.Batch.Models.PSImageReference" -ArgumentList @("UbuntuServer", "Canonical", "16.04.0-LTS", "*")
```
Comment (matthchr) [inline src/ResourceManager/AzureBatch/Commands.Batch/help/New-AzureBatchPool.md]: Given that this is powershell and we're in a Windows context, we should actually probably use windows 2016 as the example

**Comment by cormacpayne** [comment]
@matthchr any other feedback?

**Comment by matthchr** [comment]
@cormacpayne It looks good to me now

**Pull request merged**